src/Xml/Document.php

```diff
@@ -196,34 +199,247 @@ protected function getValueCollection(SimpleXMLElement $content, array $matches,
+
+                if (preg_match('/^([A-Za-z0-9_\-\.]+)\((.*)\=(.*)\)$/', $name, $matches)) {
+                    if ($name == $as) {
+                        $as = null;
+                    }
 
-                $item = $this->getSelfMatchingValue($content, $matches, $as);
+                    $item = $this->getSelfMatchingValue($content, $matches, $as);
 
-                if (is_null($as)) {
-                    $value = array_merge($value, $item);
+                    if (is_null($as)) {
+                        $value = array_merge($value, $item);
+                    } else {
+                        Arr::set($value, $as, $item);
+                    }
                 } else {
-                    Arr::set($value, $as, $item);
-                }
-            } else {
-                if ($name == '@') {
-                    $name = null;
+                    if ($name == '@') {
+                        $name = null;
+                    }
+                    Arr::set($value, $as, $this->getValue($content, $name));
                 }
+                $result = $value;
+            }
+        }
+        return $result;
+    }
 
-                Arr::set($value, $as, $this->getValue($content, $name));
+    /**
+     * Resolve values by collection of multi levels.
+     *
+     * @param  \SimpleXMLElement  $content
+     * @param  array  $uses
+     * @param  string  $objectName
+     *
+     * @return array
+     */
+    protected function parseValueCollectionMultiLevels(SimpleXMLElement $content, array $uses,$objectName)
+    {
+        $value = [];
+        $result = [];
+        $feature = $content->{$objectName};
+
+        if (!empty($feature)) {
+            foreach ($feature as $key => $contentArray) {
+                foreach ($uses as $use) {
+                    if (strpos($use, '{') !== false) {
+                        $outputSecondLevel = $this->prepareUse($use);
+                        $parent2 = $outputSecondLevel['parent'];
+                        $alias2 = $outputSecondLevel['parent_alias'];
+
+                        $value[$alias2] = $this->parseValueCollectionMultiLevels($contentArray,$outputSecondLevel['array'],$parent2);
+                    } else {
+                        list($name, $as) = strpos($use, '>') !== false ? explode('>', $use, 2) : [$use, $use];
+                        if (preg_match('/^([A-Za-z0-9_\-\.]+)\((.*)\=(.*)\)$/', $name, $matches)) {
+                            if ($name == $as) {
+                                $as = null;
+                            }
+
+                            $item = $this->getSelfMatchingValue($content, $matches, $as);
+
+                            if (is_null($as)) {
+                                $value = array_merge($value, $item);
+                            } else {
+                                Arr::set($value, $as, $item);
+                            }
```

Comment:
This clause doesn't have any test. Need to at least make sure the `preg_match` can validate expected values.

https://coveralls.io/builds/16556955/source?filename=src%2FXml%2FDocument.php

Reply:
actually this code is already exist, i don't add it. 
i just handle the case of multi levels of array, and keep the existing code,

Reply:
Ok there was bug in the code, i have fixed it and add the test case for this .
i will push when i finish :)

Reply:
The code was originally from https://github.com/orchestral/parser/issues/7 which introduce https://github.com/orchestral/parser/commit/82c29a7364749ff092dab021b7e4dc878ef7731a

<img width="1188" alt="screen shot 2018-04-18 at 6 02 19 pm" src="https://user-images.githubusercontent.com/172966/38925592-abcba3ee-4332-11e8-94a1-1ae1e7197877.png">

Reply:
i get it .
i have finished updating code . and pushed it.
but i can't see it on the pull request

Reply:
You can generate the coverage locally by running the following command.

```
./vendor/bin/phpunit --coverage-html=coverage/
```

and the `open coverage/index.html` on your browser (depending on your OS).

Reply:
i have generate coverage locally and add the test case for this code, every thing is good right now :+1: